src/status_im2/contexts/syncing/events.cljs

```diff
@@ -3,17 +3,42 @@
             [status-im2.contexts.syncing.sheets.enter-password.view :as sheet]
             [taoensso.timbre :as log]
             [utils.re-frame :as rf]
-            [utils.security.core :as security]))
+            [utils.security.core :as security]
+            [status-im2.config :as config]
+            [status-im.node.core :as node]
+            [re-frame.core :as re-frame]
+            [status-im.data-store.settings :as data-store.settings]))
+
+(defn- get-default-node-config
+  [installation-id]
+  (let [db {:networks/current-network config/default-network
+            :networks/networks        (data-store.settings/rpc->networks config/default-networks)
+            :multiaccount             {:installation-id           installation-id
+                                       :log-level                 config/log-level
+                                       :waku-bloom-filter-mode    false
+                                       :custom-bootnodes          nil
+                                       :custom-bootnodes-enabled? false}}]
+    (node/get-multiaccount-node-config db)))
 
 (rf/defn initiate-local-pairing-with-connection-string
-  {:events [:syncing/input-connection-string-for-bootstrapping]}
-  [{:keys [db]} {:keys [data]}]
-  (let [config-map        (.stringify js/JSON (clj->js {:keyUID "" :keystorePath "" :password ""}))
-        connection-string data]
-    (status/input-connection-string-for-bootstrapping
-     connection-string
-     config-map
-     #(log/info "this is response from initiate-local-pairing-with-connection-string " %))))
+  {:events       [:syncing/input-connection-string-for-bootstrapping]
+   :interceptors [(re-frame/inject-cofx :random-guid-generator)]}
+  [{:keys [random-guid-generator db]} {:keys [data]}]
+  (let [installation-id            (random-guid-generator)
+        default-node-config        (get-default-node-config installation-id)
+        default-node-config-string (.stringify js/JSON (clj->js default-node-config))
+        connection-string          data
+        callback
+        (fn [final-node-config]
+          (let [config-map (.stringify js/JSON
+                                       (clj->js {:kdfIterations         config/default-kdf-iterations
+                                                 :nodeConfig            final-node-config
+                                                 :settingCurrentNetwork config/default-network}))]
+            (status/input-connection-string-for-bootstrapping
+             connection-string
+             config-map
+             #(log/info "this is response from initiate-local-pairing-with-connection-string " %))))]
```

Comment:
FYI: the log library can do more than just concatenate strings. We can use data structures to create more meaningful messages. We can also configure the log lib with different appenders to do any sort of transformation on the output (similar to Java, just more flexible).

```clojure
#(log/info "Initiated local pairing"
           {:response %
            :event    :syncing/input-connection-string-for-bootstrapping})
```

Reply:
yeah, i remember u told me before , and i always remember it 🫶

Reply:
Oh god sorry :) I'm not gonna repeat again!